### ISSUE TITLE
Saved beatjump

### ISSUE THREAD

**Issue by acolombier** [mixxxdj/mixxx#13161]
### Feature Description

Similar to saved loop, it would be great to be able to add saved beatjump. This would allow the track to automatically jump from a preset point to another preset point. This usecase would be be to skip sections of the track with low interest (silence for example).

Currently, the workaround is to set to cues with various colour or label (eg the first one red with labelled `silence`, the second one green labelled `chorus`) and use the first red cue as visual marker, helping timing the second cue jump. This is suboptimal as:
- It uses two cues
- It is hard to perfectly time and usually leads to audio little glitch 

Like for saved loop, it would be great to be able to enable or disable the jump

**Comment by acolombier** [comment]
Indeed, I think that feature would have been covered by macros. Do I understand correctly that the macros feature was abandoned?
Could we consider saved beatjump without the whole macros framework?

**Comment by ronso0** [comment]
> Do I understand correctly that the macros feature was abandoned?

Yes, sort of. The idea is still good, even though IIUC #4528 was canceled because a) there were _some_ unedited reviews and b) Xeruf lost interest and moved on to other projects.

> Could we consider saved beatjump without the whole macros framework?

I'd say yes.
It requires a special cue type, basically like a loop cue with start and end position (where start may be after end to allow one-time backward jumps?) and do the In -> Out seek like a reverse loop.
As you described, activation would be identical to 'catching loops', just that it 'loops' once.

Setup could be pretty much the same like creating a manual loop (set In pos, seek, set Out pos).
I think as a starter, we could adopt the looping controls In / Out and add a "Jump".
Adjusting LoopingControl will be a lot of Fun :tm: :laughing: 

___
A related feature I was thinking of recently:
simplify the workflow of turing a loop cu into a regular hotcue which currently requires deleting the loop cue and create a hotcue in the same position.

**Comment by JoergAtGithub** [comment]
I just pushed a PR with an updated version of #4528 : https://github.com/mixxxdj/mixxx/pull/13172
Just the backend, no GUI.
